src/ruby.ts

```diff
@@ -0,0 +1,83 @@
+import { exec } from "child_process";
+import { promisify } from "util";
+import * as fs from "fs";
+
+import * as vscode from "vscode";
+
+const asyncExec = promisify(exec);
+const asyncReadFile = promisify(fs.readFile);
+
+export class Ruby {
+  private workingFolder: string;
+  private managerConfiguration: { [key: string]: string };
+
+  constructor() {
+    this.workingFolder = vscode.workspace.workspaceFolders![0].uri.fsPath;
+
+    this.managerConfiguration = vscode.workspace
+      .getConfiguration("rubyLsp")
+      .get("rubyVersionManager")!;
+  }
+
+  async activateRuby() {
+    switch (this.managerConfiguration.manager) {
+      case "chruby":
+        await this.activateChruby();
+        break;
+      default:
+        this.activateShadowenv();
```

Comment:
I think we want to `await` this method as well, since activate(): [Thenable](https://code.visualstudio.com/api/references/vscode-api#Thenable)<[T](https://code.visualstudio.com/api/references/vscode-api#T)>, `Extension<T>.activate` returns a promise (`Thenable<T>`) and if we return immediate, we would never be waiting for the activation to be finished.

I think that was our earlier problem and we needed to add a small delay to compensate for that.

Reply:
I've tried awaiting, but we still need the delay. There might be an `await` missing somewhere in vscode-shadowenv. I added the await anyways, but we'll keep the delay.